### PR TITLE
TYDEngine

### PR DESCRIPTION
**Pull request by lvocanson**

### CODE REVIEW

File: LightHunters/src/Const.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/// </summary>
+enum Layer {
+	Down, // Drawn under everything
+	Ground,
+	UpperGround,
+	Beings,
+	UpperBeings,
+	Decoration,
+	UpperDecoration,
+	Up, // Drawn on top of everything
```
Comment (lvocanson) [inline LightHunters/src/Const.hpp:56]: Maybe add more layers for UI

File: LightHunters/src/Const.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/// Layers for drawing.
+/// </summary>
+enum Layer {
+	Down, // Drawn under everything
+	Ground,
+	UpperGround,
+	Beings,
+	UpperBeings,
+	Decoration,
+	UpperDecoration,
```
Comment (lvocanson) [inline LightHunters/src/Const.hpp:55]: Maybe add layers for UI on top of other layers
```suggestion
	UpperDecoration,
	UiDown,
	UiBackground,
	UiLow,
	UiMid,
	UiHigh,
	UiForeground,
	UIUp,
```

File: TYDEngine/graphics/Window.cpp [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	this->clear();
+
+	if (following != nullptr)
+	{
+		sf::View view = this->getView();
+		view.setCenter(following->getPosition());
+		this->setView(view);
+	}
+	
+	for (sf::Drawable* drawable : m_renderList)
```
Comment (lvocanson) [inline TYDEngine/graphics/Window.cpp:61]: To do: Add Layer system

File: TYDEngine/graphics/Window.cpp [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+			m_mousePos = sf::Vector2i(event.mouseMove.x, event.mouseMove.y);
+			break;
+		case sf::Event::MouseWheelMoved:
+			m_mouseWheelDelta = event.mouseWheel.delta;
+			break;
+		}
+		
+	}
+	
+	// Reset mouse wheel delta if not scrolling anymore
```
Comment (lvocanson) [inline TYDEngine/graphics/Window.cpp:166]: This shouldn't be here. At the end of the function, `m_mouseWheelDelta` will always be 0.

File: TYDEngine/graphics/Window.cpp [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+	}
+
+	return false;
+}
+
+void Window::update()
+{
+	for (Updatable* u : m_updateList)
+	{
+		u->update();
```
Comment (lvocanson) [inline TYDEngine/graphics/Window.cpp:119]: We need to pass `this` in the update function, the Updatable has to ba able to call `isKeyPressed` etc.

**Review by nwiart — APPROVED** [review]
👍

**Pull request merged**